src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralServiceUnitTest.kt

```diff
@@ -795,4 +800,86 @@ class ReferralServiceUnitTest {
     assertThat(response).isSameAs(sentReferral)
     verify(referralRepository).findByIdAndSentAtIsNotNull(sentReferral.id)
   }
+
+  @Nested
+  inner class UpdateDraftReferralDetails {
+    @BeforeEach
+    fun setup() {
+      whenever(referralRepository.save(any())).thenAnswer { it.arguments[0] }
+    }
+
+    @Test
+    fun `new referral details have the same created time as the referral`() {
+      val draftReferral = referralFactory.createDraft()
+
+      // there is no existing referral details for this referral id
+      whenever(referralDetailsRepository.findLatestByReferralId(draftReferral.id)).thenReturn(null)
+
+      referralService.updateDraftReferral(
+        draftReferral,
+        DraftReferralDTO(
+          completionDeadline = LocalDate.of(3022, 3, 28)
+        )
+      )
+
+      val captor = ArgumentCaptor.forClass(ReferralDetails::class.java)
+      verify(referralDetailsRepository).saveAndFlush(captor.capture())
+
+      val referralDetails = captor.value
+      assertThat(referralDetails.createdAt).isEqualTo(draftReferral.createdAt)
+    }
+
+    @Test
+    fun `updates to the referral details for draft referrals do not create new versions`() {
+      val draftReferral = referralFactory.createDraft()
+      val existingDetails = ReferralDetails(
+        UUID.randomUUID(),
+        null,
+        draftReferral.id,
+        draftReferral.createdAt,
+        draftReferral.createdBy.id,
+        "initial version",
+        LocalDate.of(2022, 3, 28),
+        null,
+        null,
+      )
+
+      whenever(referralDetailsRepository.findLatestByReferralId(draftReferral.id)).thenReturn(existingDetails)
+      referralService.updateDraftReferral(draftReferral, DraftReferralDTO(furtherInformation = "nothing to see here"))
+
+      val captor = ArgumentCaptor.forClass(ReferralDetails::class.java)
+      verify(referralDetailsRepository).saveAndFlush(captor.capture())
+
+      val savedDetails = captor.value
+      assertThat(savedDetails.id).isEqualTo(existingDetails.id)
+      assertThat(savedDetails.createdAt).isEqualTo(existingDetails.createdAt)
+      assertThat(savedDetails.reasonForChange).isEqualTo(existingDetails.reasonForChange)
+    }
+
+    @Test
+    fun `updates to the referral details for draft referrals are saved in the existing version`() {
+      val draftReferral = referralFactory.createDraft()
+      val existingDetails = ReferralDetails(
+        UUID.randomUUID(),
+        null,
+        draftReferral.id,
+        draftReferral.createdAt,
+        draftReferral.createdBy.id,
+        "initial version",
+        LocalDate.of(2022, 3, 28),
+        null,
+        null,
+      )
+
+      whenever(referralDetailsRepository.findLatestByReferralId(draftReferral.id)).thenReturn(existingDetails)
+      referralService.updateDraftReferral(draftReferral, DraftReferralDTO(maximumEnforceableDays = 22))
```

Comment:
Is it worth testing all fields that are changeable here to ensure that all of `completionDeadline`, `furtherInformation` and `maximumEnforceableDays` are persisted?

And also, perhaps, that if they are missing (`null`) the existing values are never overwritten?

Reply:
the latter part is tested here by checking that the completion deadline is unchanged in the new version. 

i can also add furtherInformation to ensure it is also persisted.

Reply:
but i can write a new test with explicit expectations